aidial_adapter_bedrock/llm/chat_model.py

```diff
@@ -126,24 +122,33 @@ async def chat(
 
     async def truncate_prompt(
         self, params: ModelParameters, messages: List[Message]
-    ) -> List[int]:
-        prompt = self.get_text_completion_prompt(params, messages)
-        return prompt.discarded_messages or []
+    ) -> List[int] | None:
```

Comment:
> And None means that we discarded nothing
> Which is also kind of counterintuitive - why it's not just empy list then?

None and emtpy list have different meanings:
None - the algo wasn't applied at all,
Empty list - the algo was applied and the messages turned out to fit into the limit.

Here None supports optionality of the truncate prompt algo application.

So `PromptTruncationResult` isn't prompt truncation result strictly speaking, it's `OptionalPromptTruncationResult` and `PromptTruncationResult` should be a wrapper for `List[int]`. 

I really don't like proliferation of trivial types like this. Python, not Java after all.

> If you look at result type of function, and see just primitive types, that gives you no information, what meaning that types hold

Typically the name of the function provides a hint on the meaning of the result. Like

```
def concant_lists(a: List[T], b: List[T]) -> List[T]
    pass
```

It's clear what List[T] mean. You don't have to wrap it into `ConcatenationResult[T]` class.

`trucate_prompt` is literally part of the DIAL API, it not some obscure method. And it's implemented in 3 DIAL packages at this point.

Reply:
You provided great example for discussion

Because, for instance, if this function would be something like that
```python
async def truncate_prompt(
    self, params: ModelParameters, messages: List[Message]
) ->List[Message]:
    ...
```
I would have no objections - here you can get on first glance, that by `truncation` we take list of input messages, and return modified list of messages

By naming of function itself I get clearly, that truncate - is shortening/shrinking something
And this typings help me with understanding


---


Now let's look at what we have:


```python
async def truncate_prompt(
    self, params: ModelParameters, messages: List[Message]
) -> List[int] | None:
    ...
```

Ok, so I'm truncating prompt, that is list of messages, and I get ... list of integers??
What does this list of integer means? What does `None` means?

Reply:
> I really don't like proliferation of trivial types like this. It's Python, not Java after all.

If this function would be super-easy to understand, than I would have no objections.


We basically writing in language, that has 
```
Explicit is better than implicit.
...
Readability counts.
```
inside of its motto

Reply:
1. Renamed to `compute_discarded_messages`
2. Introduced `DiscardedMessages = List[int]` to improve readability
3. Added comments to explain what None means